### ISSUE TITLE
com.hazelcast.cache.impl.operation.OnJoinCacheOperationTest.test_cachePostJoinOperationSucceeds_whenJCacheAvailable_noWarningIsLogged [HZ-2555]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#23663]
_master_ (commit cd3eea7d32b161b66d32634a4c3af5213b0249ff)

Failed on zing-11: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/13/testReport/com.hazelcast.cache.impl.operation/OnJoinCacheOperationTest/test_cachePostJoinOperationSucceeds_whenJCacheAvailable_noWarningIsLogged/
<details><summary>Stacktrace:</summary>

```
Wanted but not invoked:
nodeEngine.getService(
    "hz:impl:cacheService"
);
-> at com.hazelcast.cache.impl.operation.OnJoinCacheOperationTest.test_cachePostJoinOperationSucceeds_whenJCacheAvailable_noWarningIsLogged(OnJoinCacheOperationTest.java:90)

However, there were exactly 2 interactions with this mock:
nodeEngine.getConfigClassLoader();
-> at com.hazelcast.cache.impl.operation.OnJoinCacheOperation.run(OnJoinCacheOperation.java:60)

nodeEngine.getLogger(
    class com.hazelcast.cache.impl.operation.OnJoinCacheOperation
);
-> at com.hazelcast.spi.impl.operationservice.Operation.getLogger(Operation.java:619)


	at com.hazelcast.cache.impl.operation.OnJoinCacheOperationTest.test_cachePostJoinOperationSucceeds_whenJCacheAvailable_noWarningIsLogged(OnJoinCacheOperationTest.java:90)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)

```
</details>

<details><summary>Standard output:</summary>

```
Started Running Test: sendsConfigObjectOverWire

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-ZingJDK11/13/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-11: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK11/14/testReport/junit/com.hazelcast.cache.impl.operation/OnJoinCacheOperationTest/test_cachePostJoinOperationSucceeds_whenJCacheAvailable_noWarningIsLogged/

**Comment by JamesHazelcast** [comment]
This failure is yet another example of an inconsistent Mockito failure, presenting itself after we started trialing Mockito upgrades in January 2023. This particular failure occurred on Mockito version `4.11.0`, with the "Inline Mock Maker" enabled, which was a self-described experimental mocking approach in 4.X.

We have now upgraded to Mockito version `5.3.1` and as of 5.X, the inline maker is enabled by default and more stable by nature - given that this failure only presented itself while we were using Mockito `4.11.0` with the experimental mock maker enabled (and we haven't seen a failure in the last few months), I believe there's a strong case that this failure was an anomaly caused by using Mockito 4.11.0, which we no longer use. As a result, I will close this test failure for now, awaiting another similar failure in the future to investigate further if necessary.
